packages/smart-contracts/src/contracts/ERC20SwapToPayWithConversion.sol

```diff
@@ -0,0 +1,162 @@
+pragma solidity ^0.5.12;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/ownership/Ownable.sol";
+import "./lib/SafeERC20.sol";
+import "./Erc20ConversionProxy.sol";
+import "./ChainlinkConversionPath.sol";
+
+
+interface IUniswapV2Router02 {
+  function swapTokensForExactTokens(
+    uint amountOut,
+    uint amountInMax,
+    address[] calldata path,
+    address to,
+    uint deadline
+  ) external returns (uint[] memory amounts);
+}
+
+
+/**
+ * @title ERC20SwapToPayWithConversion
+ * @notice This contract swaps ERC20 tokens before paying a request thanks to a payment proxy
+  */
+contract ERC20SwapToPayWithConversion is Ownable {
+  using SafeERC20 for IERC20;
+  using SafeMath for uint256;
+
+  IUniswapV2Router02 public swapRouter;
+  Erc20ConversionProxy public paymentProxy;
+  ChainlinkConversionPath public chainlinkConversionPath;
+
+  constructor(address _swapRouterAddress, address _paymentProxyAddress) public {
+    swapRouter = IUniswapV2Router02(_swapRouterAddress);
+    paymentProxy = Erc20ConversionProxy(_paymentProxyAddress);
+    chainlinkConversionPath = ChainlinkConversionPath(paymentProxy.chainlinkConversionPath());
+  }
+
+  /**
+  * @notice Performs a token swap between a payment currency and a payment network currency after
+  *         a conversion from a request currency and then calls a payment proxy to pay
```

Comment:
this is pretty complex. I honestly don't understand what it says

If it's too hard to make it clearer, give an example.

Reply:
Proposal:

```
Performs a request payment, denominated in a token or currency A, where the issuer expects a token B, and the payer uses a token C.
The conversion rate from A to B is done using Chainlink. 
The token swap is done using Uniswap
```